File: findCelebrity.py
# Time Complexity : O(N)
# Space Complexity : O(1)
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No


# Your code here along with comments explaining your approach
# The knows API is already defined for you.
# return a bool, whether a knows b
# def knows(a: int, b: int) -> bool:

class Solution:
    def findCelebrity(self, n: int) -> int:
        cel = 0
        
        for i in range(1, n):
            if knows(cel, i):
                cel = i
                
        for i in range(n):
            if i == cel: #can't know self
                continue
                #know each other
            if not knows(i,cel) or knows(cel, i):
                return -1
            
        return cel

File: find_celebrity.java
// Time Complexity : O(n)
// Space Complexity : O(1)



      public class Solution extends Relation {
        public int findCelebrity(int n) {
            int celebrity = 0;
            for(int i = 1; i < n; i++){
                if(knows(celebrity, i))
                    celebrity = i;
            }
            for(int i = 0; i < n; i++){
                if(i != celebrity){
                    if(!knows(i, celebrity) || knows(celebrity, i))
                        return -1;
                }
            }
            return celebrity;
        }
    }